src/canvas.class.js

```diff
@@ -492,6 +492,17 @@
      * @return {Boolean}
      */
     isTargetTransparent: function (target, x, y) {
+      if (target.shouldCache() && target._cacheCanvas && !target.isCacheDirty()) {
```

Comment:
shouldCache() returns the result of `this.objectCaching && (!this.group || this.needsItsOwnCache() || !this.group.isOnACache())`. Does this seem sufficient?